runtime/src/account_rent_state.rs

```diff
@@ -79,7 +79,7 @@ pub(crate) fn check_rent_state_with_account(
     account_state: &AccountSharedData,
 ) -> Result<()> {
     submit_rent_state_metrics(pre_rent_state, post_rent_state);
-    if !post_rent_state.transition_allowed_from(pre_rent_state) {
+    if !is_incinerator(address) && !post_rent_state.transition_allowed_from(pre_rent_state) {
```

Comment:
nit: you can do `!solana_sdk::incinerator::check_id(address)`

Reply:
Oh yeah! Duh...